### PR TITLE
Create set of bank service ap is

### LINKED ISSUE #31: Create set of `BankService` APIs
1. Table `bank_service`: id (long), name (string), price (int), periodicity (string)
2. Entity (the same fields as the table)
3. APIs: getAll, create, getById, delete

### PR DESCRIPTION
**Pull request by andy-sam**
Fixes #31

### CODE REVIEW

File: src/main/java/com/codesarge/banking/controllers/BankServiceController.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @GetMapping("bank_services")
+    public List<BankService> getAllServices() {
+        log.info("BankService list requested");
+        return bankServiceService.getAllServices();
+    }
+
+    @GetMapping("bank_services/{id}")
+    public BankService getAllServices(@PathVariable("id") Long id) {
+        log.info("BankService list requested");
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/controllers/BankServiceController.java:35]: wrong logger message

File: src/main/java/com/codesarge/banking/controllers/BankServiceController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+public class BankServiceController {
+
+    private final BankServiceService bankServiceService;
+
+    @Autowired
+    public BankServiceController(BankServiceService bankServiceService) {
+        this.bankServiceService = bankServiceService;
+    }
+
+    @GetMapping("bank_services")
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/controllers/BankServiceController.java:27]: let's replace all "bank_services" with "bank/services" in mappings

File: src/main/java/com/codesarge/banking/service/BankServiceService.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    private BankServiceRepository bankServiceRepository;
+
+    @Autowired
+    public BankServiceService(BankServiceRepository bankServiceRepository) {
+        this.bankServiceRepository = bankServiceRepository;
+    }
+
+    public List<BankService> getAllServices() {
+        return new ArrayList<>(bankServiceRepository.findAll());
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/service/BankServiceService.java:23]: you don't need to create an ArrayList. Try `return bankServiceRepository.findAll();`

File: src/main/java/com/codesarge/banking/utils/Periodicity.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package com.codesarge.banking.utils;
+
+public enum Periodicity {
+    NOT_DEFINED
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/utils/Periodicity.java:4]: let's add the following values: ONE_TIME, WEEKLY, MONTHLY

File: src/main/resources/db/changelog/changelog-2021-07-10-01.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: name
+                  type: varchar(20)
+              - column:
+                  name: price
+                  type: BIGINT
```
Comment (alex-anakin) [inline src/main/resources/db/changelog/changelog-2021-07-10-01.yaml:26]: BIGINT is too much for the price. Make it INT

File: src/main/resources/db/changelog/changelog-2021-07-10-01.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+              - column:
+                  name: id
+                  type: BIGINT
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: name
+                  type: varchar(20)
```
Comment (alex-anakin) [inline src/main/resources/db/changelog/changelog-2021-07-10-01.yaml:23]: What if a service needs to have a long name? Use varchar(50) here.

**Review by alex-anakin — APPROVED** [review]
I'm approving it because the required changes are minor. Please fix and resolve comments. Then you will be able to merge on your own if there are no git conflicts. If you get conflicts, try to resolve them. And merge afterward.

File: src/main/java/com/codesarge/banking/controllers/BankServiceController.java (hunk 1 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/controllers/BankServiceController.java:35]: done

File: src/main/java/com/codesarge/banking/controllers/BankServiceController.java (hunk 2 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/controllers/BankServiceController.java:27]: done

File: src/main/java/com/codesarge/banking/service/BankServiceService.java (hunk 3 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/service/BankServiceService.java:23]: done

File: src/main/java/com/codesarge/banking/utils/Periodicity.java (hunk 4 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/utils/Periodicity.java:4]: done

File: src/main/resources/db/changelog/changelog-2021-07-10-01.yaml (hunk 5 above)
Comment (andy-sam) [reply src/main/resources/db/changelog/changelog-2021-07-10-01.yaml:26]: done

File: src/main/resources/db/changelog/changelog-2021-07-10-01.yaml (hunk 6 above)
Comment (andy-sam) [reply src/main/resources/db/changelog/changelog-2021-07-10-01.yaml:23]: done

**Pull request merged**